app/services/submission/trait_score_parser.rb

```diff
@@ -34,16 +34,20 @@ def call
 
   def map_headers_to_traits
     @upload.log "Mapping file header columns to Trait Descriptors"
-    header = input.readline
+    header = csv.readline
     @trait_names = PlantTrialSubmissionDecorator.decorate(@upload.submission).sorted_trait_names
     @trait_mapping = {}
-    header.split("\t")[1..-1].each_with_index do |column_name, i|
-      next if i >= @trait_names.length
-      column_name = column_name.strip
-      trait_index = @trait_names.find_index(column_name)
-      trait_index = i unless trait_index.present?
-      @trait_mapping[i] = trait_index
-      @upload.log " - Mapped column '#{column_name}' to Trait index #{trait_index} of value #{@trait_names[trait_index]}"
+    if header.blank?
```

Comment:
Is this even possible? I think only when an empty file is uploaded, but then the "Choose file..." button remains disabled and nothing happens, as paperclip detects content-type spoofing.

`[paperclip] Content Type Spoof: Filename empty.csv (["text/comma-separated-values", "text/csv"]), content type discovered from file command: application/octet-stream. See documentation to allow this combination.
`

I think we can check if the first line begins with 'Plant scoring unit name' and assume the file is not formatted correctly otherwise.